src/org/zalando/stups/friboo/system/http.clj

```diff
@@ -11,41 +11,141 @@
+            [io.sarnowski.swagger1st.util.api :as s1stapi]
             [ring.adapter.jetty :as jetty]
-            [ring.util.response :as r]
-            [ring.middleware.resource :refer [wrap-resource]]
-            [ring.middleware.content-type :refer [wrap-content-type]]
-            [ring.middleware.not-modified :refer [wrap-not-modified]]
-            [ring.middleware.gzip :refer [wrap-gzip]]
-            [clojure.data.codec.base64 :as b64]
-            [io.clj.logging :refer [with-logging-context]]
-            [clj-http.client :as client]
-            [com.netflix.hystrix.core :refer [defcommand]]
-            [clojure.core.memoize :as memo]
-            [clojure.core.cache :as cache]
-            [clojure.core.incubator :refer [dissoc-in]])
+            [org.zalando.stups.friboo.security :as security]
+            [ring.middleware.gzip :as gzip]
+            [com.stuartsierra.component :refer [Lifecycle]])
   (:import (com.netflix.hystrix.exception HystrixRuntimeException)
-           (org.zalando.stups.txdemarcator Transactions)))
+           (org.zalando.stups.txdemarcator Transactions)
+           (clojure.lang ExceptionInfo)))
 
-(defn flatten-parameters
+(defn merged-parameters
   "According to the swagger spec, parameter names are only unique with their type. This one assumes that parameter names
    are unique in general and flattens them for easier access."
   [request]
   (apply merge (vals (:parameters request))))
 
+(defn make-resolver-fn [controller]
+  "Calls operationId function with controller, flattened request params and raw request map."
+  (fn [request-definition]
+    (when-let [operation-fn (s1stexec/operationId-to-function request-definition)]
+      (fn [request]
+        (operation-fn controller (merged-parameters request) request)))))
+
+(defn middleware-chain [context middlewares]
+  (reduce #(%2 %1) context middlewares))
+
+(defn flatten1
+  "Flattens the collection one level, for example, converts {:a 1 :b 2} to (:a 1 :b 2)."
+  [coll]
+  (apply concat coll))
+
+(defn- with-flattened-options-map
+  "If f is defined like this:
+
+  `(defn f [first-arg & {:keys [a b] :as opts}] ... )`
+
+  makes it convenient to pass `opts` as a map:
+
+  `(-> first-arg (with-flattened-options-map f {:a 1 :b 2}))`
+  will result in
+  `(f first-arg :a 1 :b 2)`
+
+  Used in `start-component` for clarity.
+  "
+  [first-arg f options]
+  (apply f first-arg (flatten1 options)))
+
+(defn start-component [{:as this :keys [api-resource configuration middlewares security-handlers controller s1st-options]}]
+  (log/info "Starting HTTP daemon for API %s" api-resource)
+  (when (:handler this)
+    (throw (ex-info "Component already started, aborting." {})))
+  ;; Refer to default-middlewares object below for middleware lists
+  ;; Create context from the YAML definition on the classpath, optionally provide validation flag
+  (let [handler (-> (apply s1st/context :yaml-cp api-resource (flatten1 (:context s1st-options)))
```

Comment:
// If I wrote swagger1st I'd have option maps as maps to not convert them forth and back in every place. In particular, it would streamline this piece substantially.